### ISSUE TITLE
StackExchange.Redis 2.1.58 with .NET framework 4.62 (net462)

### ISSUE THREAD

**Issue by Vikakuma** [StackExchange/StackExchange.Redis#1592]
The NuGet dependencies listed in the StackExchange.Redis nuget package don't actually satisfy StackExchange.Redis dll's assembly dependencies (for .NET 461 or .NET 462)

The assembly StackExchange.Redis 2.0.0.0 depends on System.IO.Compression 4.2.0.0. Looking at the StackExchange.Redis package (nuget version 2.1.58) I see that it lists System.IO.Compression (nuget version >= 4.3.0) as a dependency. If I download the System.IO.Compression package and look at the System.IO.Compression assembly contained within, the version I see is 4.1.2.0, not 4.2.0.0.

This is also true for System.Runtime.InteropServices.RuntimeInformation. StackExchange.Redis.dll depends on System.Runtime.InteropServices.RuntimeInformation.dll 4.0.2.0, but the nuget package referenced by StackExchange.Redis contains System.Runtime.InteropServices.RuntimeInformation.dll 4.0.1.0.

**Comment by NickCraver** [comment]
I think you're considering NuGet versions vs. DLL versions, but there are often not the same thing. If you're seeing a dependency mismatch here, you'll want to ensure you have the latest NuGet tooling, since earlier versions had some trouble here.

**Comment by Vikakuma** [comment]
Thanks for reply @NickCraver 
No, I m checking the assembly version present inside the NuGet package.

For e.g. 
StackExchange.Redis.2.1.58 depends on System.IO.Compression >= 4.3.0 (NuGet version) but the dll version inside this NuGet is 4.1.2.0 and NOT 4.2.0.0 as mentioned in assembly manifest of StackExchange.Redis.2.1.158

Here's the image of manifest file of StackExchange.Redis.2.1.58
![image](https://user-images.githubusercontent.com/73263889/96817251-ce979580-13d3-11eb-93e2-cf6c17b5fd07.png)

**Comment by mgravell** [comment]
Ultimately, all *we're* doing here is including package references in the build, specifically:

```
<PackageReference Update="System.IO.Compression" Version="4.3.0" />
```

and 

```
<PackageReference Include="System.IO.Compression" Condition="'$(TargetFramework)' == 'net461' or '$(TargetFramework)' == 'net472'" />
```

I'm open to suggestions on what we can do if the build tools are doing something unusual or bizarre with this binding...

**Comment by mgravell** [comment]
Working on this here: https://github.com/StackExchange/StackExchange.Redis/pull/1594

However, and I **cannot express this loudly enough**: net46whatever is *basically* dead, and it is becoming increasingly hard to support. I strongly strongly advise getting "move to .NET Core" onto your roadmap. It *will* be dropped eventually (I can see us making netcoreapp3.1 the minimum baseline); for more context, see: https://blog.marcgravell.com/2020/01/net-core-net-5-exodus-of-net-framework.html

**Comment by NickCraver** [comment]
Resolved in #1594, closing out!
